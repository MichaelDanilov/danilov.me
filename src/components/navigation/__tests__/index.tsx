import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '..';

import Languages from '../../../constants/languages';

describe('Navigation', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with some navigation', () => {
    const tree = renderer
      .create(
        <Navigation
          navigation={[
            {
              title: 'English version',
              href: '/en',
              lang: Languages.english,
            },
            {
              title: 'Basic version',
              href: '/',
            },
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
