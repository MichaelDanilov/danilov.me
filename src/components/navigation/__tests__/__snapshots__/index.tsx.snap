// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<ul
  className="sc-bdVaJa hTzUzq"
/>
`;

exports[`Navigation with some navigation 1`] = `
<ul
  className="sc-bdVaJa hTzUzq"
>
  <li
    className="sc-bwzfXH dmRCrB"
  >
    <a
      href="/en"
      hrefLang="en"
      lang="en"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      English version
    </a>
  </li>
  <li
    className="sc-bwzfXH dmRCrB"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      Basic version
    </a>
  </li>
</ul>
`;
